### PR TITLE
Refactor and document subtract()

### PR DESCRIPTION
**Pull request by trosi**
Rewrites subtract() using simpler and more generic argument names and introducing minimal type checking. 

Also adds documentation for the same function.

### CODE REVIEW

**Comment by jonocarroll** [comment]
Thanks, and welcome!

There's a lot that's good here; you've explained what you've proposed and the additional documentation looks good. 

This is of course a toy example, but pretending that this was a real project:
 - did you notice that you swapped the argument order when refactoring? these were of course the 'wrong' way around to start with, but it is the sort of detail that should be brought to attention. A refactor shouldn't change behaviour. A bugfix should.
 - the original version (incorrectly) used super-assignment (`<<-`) which you've changed to regular assignment. Again, that was 'wrong' to start with, but perhaps there was a reason for it you haven't taken into account. Mentioning it helps the project owner take on the ownership of that change.

When proposing a change I like to run `R CMD check` or `devtools::check()` or use the shortcut in an IDE to see if the package is currently passing, then run it again to ensure my proposed changes don't make anything worse. This package certainly has a lot that's failing, but this PR introduces a new one; `@example` code should only have the _code_, not the output, because all of it will be evaluated. In this case, your example throws an error

```r
checking for unstated dependencies in examples ... ERROR
  Warning: parse error in file 'lines':
  1: unexpected '>'
  36: subtract(c(10, 10), c(7, 20))
  37: >
      ^
```

On the plus side, that means there's something I can ask you to change. I'll start a proper review so you can see what that looks like, then you need to push another commit to your branch.

**Review by jonocarroll — CHANGES_REQUESTED** [review]
Please take this as positive feedback; I'm nitpicking to provide the opportunity for you to try out the review process.

File: R/multiply.R [hunk 1]
```diff
@@ -3,9 +3,22 @@ add <- function(firstvariablename = "7", secondvariablename) {
...
+#'
+#' @param x A numeric vector from which \code{y} will be subtracted.
+#' @param y A numeric vector to subtract.
+#'
+#' @return A numeric vector with the difference between \code{x} and \code{y}.
+#' @export
+#'
+#' @examples
+#' subtract(c(10, 10), c(7, 20))
+#' > [1]  3 -10
```
[the remark this answers is not in the document]
Comment (trosi) [reply R/multiply.R]: Thanks! I have fixed this in e7c42a4091890a2edc00c88eff0855eeef65b31b

File: R/multiply.R [hunk 2]
```diff
@@ -3,9 +3,22 @@ add <- function(firstvariablename = "7", secondvariablename) {
...
+subtract <- function(x, y) {
+  stopifnot(is.numeric(x), is.numeric(y))
```
[the remark this answers is not in the document]
Comment (trosi) [reply R/multiply.R:19]: Indeed, in retrospective I think the type test might be doing more harm than good! After all, `x - y` already fails with a similar error message if subtraction is not defined. Thanks for the comment, I was not thinking of complex numbers.

**Comment by trosi** [comment]
By the way, is it correct to fork and keep working off the main branch? Or would it be better to switch to a new branch? Thanks again.

**Comment by jonocarroll** [comment]
This looks good - I'm going to merge it. 

FYI, you can edit your top-level comment if you need to add more details.

In terms of working with branches see some discussions in the other PRs. In general, using your main branch is usually fine, but if your pull request doesn't get merged for some reason then it won't be in the same state as the original, so I try to make my changes in a new branch of my fork, and occasionally sync my fork's main branch with the original (GitHub now has a button for this).

Working in a branch also means that you retain a forked copy of the original that you haven't changed, in case you want to make some local comparisons.

If you create a pull request via the web editor it creates a 'patch-1' (or something similar) branch in your fork. If and when the PR is merged you can safely delete that feature branch and sync the main branch.

**Comment by jonocarroll** [comment]
Congrats on a successful pull request! 🎉

**Comment by trosi** [comment]
Awesome! Many thanks again, this was extremely helpful

**Pull request merged**